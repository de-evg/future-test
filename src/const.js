export const showingStatus = {
    UNSET: `UNSET`,
    SHOW: `SHOW`,
    HIDE: `HIDE`
};

export const DEFAULT_SORT = `ID_UP`;

export const SortType = {
    ID_UP: `ID_UP`,
    ID_DOWN: `ID_DOWN`,
    FIRSTNAME_UP: `FIRSTNAME_UP`,
    FIRSTNAME_DOWN: `FIRSTNAME_DOWN`,
    LASTNAME_UP: `LASTNAME_UP`,
    LASTNAME_DOWN: `LASTNAME_DOWN`,
    EMAIL_UP: `EMAIL_UP`,
    EMAIL_DOWN: `EMAIL_DOWN`,
    PHONE_UP: `PHONE_UP`,
    PHONE_DOWN: `PHONE_DOWN`,
};