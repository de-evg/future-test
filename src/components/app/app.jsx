import React from "react";
import PropTypes from "prop-types";
import Table from "../table/table";
import Search from "../search/search";
import MainTableBody from "../main-table-body/main-table-body";
import AddTableForm from "../add-table-form/add-table-form";
import MainMenu from "../main-menu/main-menu";
import { connect } from "react-redux";
import { NameSpace } from "../../store/reducers/root";
import { showingStatus } from "../../const";
import TableHead from "../main-table-head/main-table-head";
import Pagination from "../pagination/pagination";
import Loader from "../loader/loader";

const App = ({
  addRowFormVisualStatus,
  searchFormVisualStatus,
  isLoading,
  users,
}) => {
  let visualClass = "";
  if (
    addRowFormVisualStatus === showingStatus.HIDE ||
    searchFormVisualStatus === showingStatus.HIDE
  ) {
    visualClass = "container--hidden";
  }
  if (
    addRowFormVisualStatus === showingStatus.SHOW ||
    searchFormVisualStatus === showingStatus.SHOW
  ) {
    visualClass = "container--show";
  }
  return (
    <>
      <MainMenu />
      <div className={`container ${visualClass}`}>
        <AddTableForm />
        <Search />
      </div>
      {isLoading && <Loader />}
      {!isLoading && !!users.length && (
        <>
          <Table TableHead={<TableHead />} TableBody={<MainTableBody />} />
          <Pagination />
        </>
      )}
    </>
  );
};

App.propTypes = {
  addRowFormVisualStatus: PropTypes.string.isRequired,
  searchFormVisualStatus: PropTypes.string.isRequired,
  users: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
      firstName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
      phone: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired,
    })
  ),
  isLoading: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  addRowFormVisualStatus:
    state[NameSpace.INTERFACE_DATA].addRowFormVisualStatus,
  searchFormVisualStatus:
    state[NameSpace.INTERFACE_DATA].searchFormVisualStatus,
  users: state[NameSpace.USERS_DATA].users,
  isLoading: state[NameSpace.INTERFACE_DATA].isLoading,
});

export { App };
export default connect(mapStateToProps)(App);
