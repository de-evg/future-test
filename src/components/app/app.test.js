import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {shallowToJson} from 'enzyme-to-json';
import { showingStatus } from "../../const";
import { App } from "./app";

Enzyme.configure({adapter: new Adapter()});

describe(`render App`, () => {
  it(`Should App render correctly with UNSET showing status, FALSE isLoading`, () => {
    const wrapper = shallow(<App
      addRowFormVisualStatus={showingStatus.UNSET}
      searchFormVisualStatus={showingStatus.UNSET}
      isLoading={false}
      users={[]}
    />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it(`Should App render correctly with HIDE showing status, TRUE isLoading`, () => {
    const wrapper = shallow(<App
      addRowFormVisualStatus={showingStatus.HIDE}
      searchFormVisualStatus={showingStatus.HIDE}
      isLoading={true}
      users={[]}
    />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  
  it(`Should App render correctly with SHOW showing status, TRUE isLoading`, () => {
    const wrapper = shallow(<App
      addRowFormVisualStatus={showingStatus.SHOW}
      searchFormVisualStatus={showingStatus.SHOW}
      isLoading={true}
      users={[]}
    />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
