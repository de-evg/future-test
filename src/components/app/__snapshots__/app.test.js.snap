// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App Should App render correctly with HIDE showing status, TRUE isLoading 1`] = `
<Fragment>
  <MainMenu />
  <div
    className="container container--hidden"
  >
    <Connect(AddTableForm) />
    <Connect(Search) />
  </div>
  <Loader />
</Fragment>
`;

exports[`render App Should App render correctly with SHOW showing status, TRUE isLoading 1`] = `
<Fragment>
  <MainMenu />
  <div
    className="container container--show"
  >
    <Connect(AddTableForm) />
    <Connect(Search) />
  </div>
  <Loader />
</Fragment>
`;

exports[`render App Should App render correctly with UNSET showing status, FALSE isLoading 1`] = `
<Fragment>
  <MainMenu />
  <div
    className="container "
  >
    <Connect(AddTableForm) />
    <Connect(Search) />
  </div>
</Fragment>
`;
